lab2.js

```diff
@@ -121,26 +143,35 @@ assert(human.sayHello(klingon) === 'nuqneH',
 // assertions for each one
 //*********************************************************
 function max(array) {
+  return Math.max.apply(Math, array);
   // TODO: return the largest number in the given array
 }
 
 // TODO: write three more assertions
-assert(max([ 1, 3, 2 ]) === 3, '[1,3,2]');
+assert(max([1, 3, 2]) === 3, '[1,3,2]');
+assert(max([40, 2, 1.9]) === 40, '[40, 2, 1.9]');
+assert(max(['dog', 'cat', 'ant']) === 'dog', '[\'dog\', \'cat\', \'ant\']');
+assert(max([100, 10.0, 1.00]) === 100, '[100, 10.0, 1.00]');
 
 function variablify(string) {
-  // TODO: you are given a string with several words in it
-  // return a corresponding variable name that follows
-  // javascript conventions
-  // HINT:
-  // you might want to use these string methods:
-  //  split(), charAt(), toUpperCase()
-  // and this array method: join()
+  var splitString = string.split(' ');
+  var rightCase = [];
+  for (var i = 1; i < splitString.length; i++) {
+    var newString = splitString[i].charAt(0).toUpperCase() + splitString[i].substr(1);
+    rightCase.push(newString);
+  }
+  rightCase.unshift(splitString[0]);
+  rightCase.join('');
```

Comment:
.join doesn't modify the rightCase array -  it returns a new array. Since you're not "catching" that returned array anywhere (e.g. in a variable), then you're just returning an array out of this function - and that's why the asserts are failing. You'll need to do something like 
var variableName = rightCase.join('');
and then return variableName. Or you can just return
rightCase.join('')
directly.